node-js-basics-101/node_modules/asynquence-contrib/contrib-es6.src.js

```diff
@@ -0,0 +1,2360 @@
+				tmp = arr;
+				arr = each;
+				each = tmp;
+			}
+
+			return ASQ.apply(ø,args)
+			.gate.apply(ø,arr.map(function $$map(item){
+				return function $$segment(){
+					each.apply(ø,[item].concat(ARRAY_SLICE.call(arguments)));
+				};
+			}));
+		})
+		.val(function $$val(){
+			// collect all gate segment output into one value-message
+			// Note: return a normal array here, not a message wrapper!
+			return ARRAY_SLICE.call(arguments);
+		});
+
+		return api;
+	};
+});
+// "none"
+ASQ.extend("none",function $$extend(api,internals){
+	return function $$none() {
+		if (internals("seq_error") || internals("seq_aborted") ||
+			arguments.length === 0
+		) {
+			return api;
+		}
+
+		var fns = ARRAY_SLICE.call(arguments);
+
+		api.then(function $$then(done){
+			function reset() {
+				finished = true;
+				error_messages.length = 0;
+				success_messages.length = 0;
+			}
+
+			function complete(trigger) {
+				if (success_messages.length > 0) {
+					// any successful segment's message(s) sent
+					// to main sequence to proceed as **error**
+					success_messages.length = fns.length;
+					trigger.fail.apply(ø,success_messages);
+				}
+				else {
+					// send errors as **success** to main sequence
+					error_messages.length = fns.length;
+					trigger.apply(ø,error_messages);
+				}
+
+				reset();
+			}
+
+			function success(trigger,idx,args) {
+				if (!finished) {
+					completed++;
+					success_messages[idx] =
+						args.length > 1 ?
+						ASQ.messages.apply(ø,args) :
+						args[0]
+					;
+
+					// all segments complete?
+					if (completed === fns.length) {
+						finished = true;
+
+						complete(trigger);
+					}
+				}
+			}
+
+			function failure(trigger,idx,args) {
+				if (!finished &&
+					!(idx in error_messages)
+				) {
+					completed++;
+					error_messages[idx] =
+						args.length > 1 ?
+						ASQ.messages.apply(ø,args) :
+						args[0]
+					;
+				}
+
+				// all segments complete?
+				if (!finished &&
+					completed === fns.length
+				) {
+					finished = true;
+
+					complete(trigger);
+				}
+			}
+
+			var completed = 0, error_messages = [], finished = false,
+				sq = ASQ.apply(ø,ARRAY_SLICE.call(arguments,1)),
+				success_messages = []
+			;
+
+			wrapGate(sq,fns,success,failure,reset);
+
+			sq.pipe(done);
+		});
+
+		return api;
+	};
+});
+// "pThen"
+ASQ.extend("pThen",function $$extend(api,internals){
+	return function $$pthen(success,failure) {
+		if (internals("seq_aborted")) {
+			return api;
+		}
+
+		var ignore_success_handler = false, ignore_failure_handler = false;
+
+		if (typeof success === "function") {
+			api.then(function $$then(done){
+				if (!ignore_success_handler) {
+					var ret, msgs = ASQ.messages.apply(ø,arguments);
+					msgs.shift();
+
+					if (msgs.length === 1) {
+						msgs = msgs[0];
+					}
+
+					ignore_failure_handler = true;
+
+					try {
+						ret = success(msgs);
+					}
+					catch (err) {
+						if (!ASQ.isMessageWrapper(err)) {
+							err = [err];
```

Comment:
__[jshint]__
* Do not assign to the exception parameter.

<!-- Comment by [SideCI](https://sideci.com) -->